src/order/filter/basic.lean

```diff
@@ -1699,6 +1704,34 @@ instance ultrafilter.has_bind : has_bind ultrafilter := ⟨@ultrafilter.bind⟩
 instance ultrafilter.functor : functor ultrafilter := { map := @ultrafilter.map }
 instance ultrafilter.monad : monad ultrafilter := { map := @ultrafilter.map }
 
+noncomputable def hyperfilter : filter α := ultrafilter_of cofinite
+
+lemma hyperfilter_le_cofinite (hi : set.infinite (@set.univ α)) : @hyperfilter α ≤ cofinite := 
+  (ultrafilter_of_spec (cofinite_ne_bot hi)).1
+
+lemma is_ultrafilter_hyperfilter (hi : set.infinite (@set.univ α)) : is_ultrafilter (@hyperfilter α) := 
+  (ultrafilter_of_spec (cofinite_ne_bot hi)).2
+
+theorem not_hyper_of_finite (hi : set.infinite (@set.univ α)) {s : set α} :
+set.finite s → s ∉ (@hyperfilter α).sets :=
+  λ hf hy, 
+  have hx : -s ∉ hyperfilter.sets := λ hs, 
+    (ultrafilter_iff_compl_mem_iff_not_mem.mp (is_ultrafilter_hyperfilter hi) s).mp hs hy,
+  have ht : -s ∈ cofinite.sets := by show -s ∈ {s | _}; 
+    rwa [set.mem_set_of_eq, lattice.neg_neg],
+  hx $ hyperfilter_le_cofinite hi ht
+
+theorem hyper_of_cofinite (hi : set.infinite (@set.univ α)) {s : set α} :
+set.finite s → -s ∈ (@hyperfilter α).sets :=
+  λ hf, (ultrafilter_iff_compl_mem_iff_not_mem.mp 
+    (is_ultrafilter_hyperfilter hi) s).mpr $ 
+    not_hyper_of_finite hi hf
+
+theorem hyper_of_cofinite' (hi : set.infinite (@set.univ α)) {s : set α} :
```

Comment:
Name: `mem_hyperfilter_of_finite_compl`
Remove `sets`, indentation, add `(hs : set.finite (-s))` instead of `λ hf, `